--- a/src/tuxedo_compatibility_check/tuxedo_compatibility_check.c
+++ b/src/tuxedo_compatibility_check/tuxedo_compatibility_check.c
@@ -216,6 +216,13 @@
 			DMI_MATCH(DMI_CHASSIS_VENDOR, "TUXEDO"),
 		},
 	},
+	{
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "MECHREVO"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "WUJIE14XA"),
+			DMI_MATCH(DMI_BOARD_NAME, "WUJIE14-GX4HRXL"),
+		},
+	},
 	{ }
 };

--- a/src/uniwill_leds.h
+++ b/src/uniwill_leds.h
@@ -252,6 +252,13 @@
 	},
 	{
 		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "MECHREVO"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "WUJIE14XA"),
+			DMI_MATCH(DMI_BOARD_NAME, "WUJIE14-GX4HRXL"),
+		},
+	},
+	{
+		.matches = {
 			DMI_MATCH(DMI_SYS_VENDOR, "TUXEDO"),
 			DMI_MATCH(DMI_BOARD_NAME, "XxHP4NAx"),
 		},

--- a/src/uniwill_keyboard.h
+++ b/src/uniwill_keyboard.h
@@ -1789,6 +1789,13 @@
 	return false;
 }
 
+static bool uniwill_is_mechrevo_wujie14xa(void)
+{
+	return dmi_match(DMI_SYS_VENDOR, "MECHREVO") &&
+		dmi_match(DMI_PRODUCT_NAME, "WUJIE14XA") &&
+		dmi_match(DMI_BOARD_NAME, "WUJIE14-GX4HRXL");
+}
+
 static int uniwill_keyboard_probe(struct platform_device *dev)
 {
 	u32 i;
@@ -1803,7 +1810,8 @@
 	// FIXME Hard set balanced profile until we have implemented a way to
 	// switch it while tuxedo_io is loaded
 	// uw_ec_write_addr(0x51, 0x07, 0x00, 0x00, &reg_write_return);
-	uniwill_write_ec_ram(0x0751, 0x00);
+	if (!uniwill_is_mechrevo_wujie14xa())
+		uniwill_write_ec_ram(0x0751, 0x00);
 
 	if (uw_feats->uniwill_profile_v1) {
 		// Set manual-mode fan-curve in 0x0743 - 0x0747
@@ -1819,10 +1827,12 @@
 	uniwill_set_custom_profile_mode(true);
 
 	// Enable manual mode
-	uniwill_write_ec_ram(0x0741, 0x01);
+	if (!uniwill_is_mechrevo_wujie14xa())
+		uniwill_write_ec_ram(0x0741, 0x01);
 
 	// Zero second fan temp for detection
-	uniwill_write_ec_ram(0x044f, 0x00);
+	if (!uniwill_is_mechrevo_wujie14xa())
+		uniwill_write_ec_ram(0x044f, 0x00);
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(5, 9, 0)
 	TUXEDO_ERROR("Warning: Kernel version less that 5.9, keyboard backlight might not be properly recognized.");
